### ISSUE TITLE
Opening window with python 3 fails under Linux with pyglet, but not with pygame.

### ISSUE THREAD

**Issue by igordertigor** [psychopy/psychopy#1982]
Hello,
I'm using psychopy with python 3 (3.6 on Ubuntu 18.04, same issue was with py3.5 on Ubuntu 16.04). When I try to open a window I get a pretty intricate error message. Here is a minimal code example:
```python
from psychopy import visual
win = visual.Window()
```
And here is the error message:
```
pygame 1.9.4
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "minimal.py", line 3, in <module>
    win = visual.Window()
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/window.py", line 375, in __init__
    self.backend = backends.getBackend(win=self, *args, **kwargs)
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/backends/__init__.py", line 32, in getBackend
    return Backend(win, *args, **kwargs)
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/backends/pygletbackend.py", line 227, in __init__
    self._origGammaRamp = self.getGammaRamp()
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/backends/pygletbackend.py", line 326, in getGammaRamp
    return getGammaRamp(self.screenID, self.xDisplay)
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/backends/gamma.py", line 120, in getGammaRamp
    rampSize = getGammaRampSize(screenID, xDisplay=xDisplay)
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/backends/gamma.py", line 285, in getGammaRampSize
    raise RuntimeError("Gamma ramp size is reported as 0.")
RuntimeError: Gamma ramp size is reported as 0.
0.4679 	WARNING 	Monitor specification not found. Creating a temporary one...
Exception ignored in: <bound method Window.__del__ of <psychopy.visual.window.Window object at 0x7fbf8319f2b0>>
Traceback (most recent call last):
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/window.py", line 478, in __del__
    self.close()
  File "/home/ingo/tmp/psychopy_dep/.venv3/lib/python3.6/site-packages/psychopy/visual/window.py", line 1124, in close
    self.backend.close()  # moved here, dereferencing the window prevents
AttributeError: 'NoneType' object has no attribute 'close'
```
It is possible to solve the issue by explicitly setting `winType='pygame'`, i.e. the following works:
```python
from psychopy import visual
win = visual.Window(winType='pygame')
```
That doesn't seem to be a clever long term strategy. In particular given that (i) tests of visual stimuli with pygame are commented out and that (ii) there are open discussions (#1280) to potentially remove pygame support from python altogether. Is this a known issue? Are there established solutions for it? Do others experience the same?

**Comment by djmannion** [comment]
The key line in the error message is ```RuntimeError: Gamma ramp size is reported as 0.```. This is due to some video cards (Intel, typically) not showing the expected behaviour (see [here](https://discourse.psychopy.org/t/gamma-problem-in-v1-90-2/4549/4)). It was resolved for me by installing specific intel video drivers in my operating system - the same approach may work in ubuntu.

**Comment by djmannion** [comment]
I've raised that option a couple of times ([in the original RFC](https://github.com/psychopy/psychopy/pull/1626#issuecomment-360957903) and [subsequently](https://discourse.psychopy.org/t/gamma-problem-in-v1-90-2/4549/4?u=djmannion)).

If the response is now "surely" for a warning, then that sounds fine to me.

**Comment by peircej** [comment]
Ha! Sorry for missing this in our suggestions before @djmannion :-) I think the answer remains "surely"

**Comment by igordertigor** [comment]
@piercej: Do I understand correctly, that 1.90 does not include glfw yet? When I use version 3.0.0b2, I can open a window. So this error seems to be solved. I haven't had time to play around much with the new version though. Also, congratulations on speeding up the pip-installation. wxPython took ages to compile...

**Comment by peircej** [comment]
@igordertigor watch out for my name - it's unusual in being e before i!

I think the first iteration of the glfw window was actually included in 1.90 but it was just an infant and a lot of things didn't work (e.g. no TextStim)
